core/trino-spi/src/main/java/io/trino/spi/classloader/ThreadContextClassLoader.java

```diff
@@ -13,22 +13,49 @@
  */
 package io.trino.spi.classloader;
 
-import java.io.Closeable;
-
 public class ThreadContextClassLoader
-        implements Closeable
 {
-    private final ClassLoader originalThreadContextClassLoader;
+    private ThreadContextClassLoader() {}
+
+    public static void withClassLoaderOf(Class<?> clazz, Runnable runnable)
+    {
+        withClassLoader(clazz.getClassLoader(), runnable);
+    }
+
+    public static <T, E extends Exception> T withClassLoaderOf(Class<?> clazz, ThrowingSupplier<T, E> supplier)
+    {
+        return withClassLoader(clazz.getClassLoader(), supplier);
+    }
+
+    public static void withClassLoader(ClassLoader newThreadContextClassLoader, Runnable runnable)
+    {
+        withClassLoader(newThreadContextClassLoader, () -> {
+            runnable.run();
+            return null;
+        });
+    }
 
-    public ThreadContextClassLoader(ClassLoader newThreadContextClassLoader)
```

Comment:
This is SPI breaking change. Deprecate, but don't remove.

Or even don't deprecate, since the new version is _almost_ a drop-in replacement (multiple checked exceptions)

Reply:
done, original implementation restored